File: data/code_pairs/cpp/2232.cpp
#include <cmath>

void iau_PAS(double AL, double AP, double BL, double BP, double &THETA) {
    double DL, X, Y;

    DL = BL - AL;
    Y = std::sin(DL) * std::cos(BP);
    X = std::sin(BP) * std::cos(AP) - std::cos(BP) * std::sin(AP) * std::cos(DL);

    if (X != 0.0 || Y != 0.0) {
        THETA = std::atan2(Y, X);
    } else {
        THETA = 0.0;
    }
}

File: data/code_pairs/fortran/2232.f90
SUBROUTINE iau_PAS(AL, AP, BL, BP, THETA)
    IMPLICIT NONE
    DOUBLE PRECISION, INTENT(IN) :: AL, AP, BL, BP
    DOUBLE PRECISION, INTENT(OUT) :: THETA
    DOUBLE PRECISION :: DL, X, Y

    DL = BL - AL
    Y = SIN(DL) * COS(BP)
    X = SIN(BP) * COS(AP) - COS(BP) * SIN(AP) * COS(DL)

    IF (X /= 0D0 .OR. Y /= 0D0) THEN
        THETA = ATAN2(Y, X)
    ELSE
        THETA = 0D0
    END IF
END SUBROUTINE iau_PAS